qiskit/circuit/library/arithmetic/multipliers/multiplier.py

```diff
@@ -0,0 +1,98 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Compute the product of two equally sized qubit registers."""
+
+from typing import Optional
+
+from qiskit.circuit import QuantumCircuit
+
+
+class Multiplier(QuantumCircuit):
+    r"""Compute the product of two equally sized qubit registers into a new register.
+
+    For two input registers :math:`|a\rangle_n`, :math:`|b\rangle_n` with :math:`n` qubits each
+    and an output register with :math:`2n` qubits, a multiplier performs the following operation
+
+    .. math::
+
+        |a\rangle |b\rangle |0\rangle \mapsto |a\rangle |b\rangle |a \cdot b\rangle
```

Comment:
Should we add register sizes here as well, since they are in all other equations?
```suggestion
        |a\rangle_n |b\rangle_n |0\rangle_{t} \mapsto |a\rangle_n |b\rangle_n |a \cdot b\rangle_t
        
where :math:`t` is the number of bits used to represent the result. To completely store the result
of the multiplication without overflow we need :math:`t = 2n` bits. 
```

Reply:
Makes sense, thanks for this suggestion 🙂